sdlf-utils/ingestion-examples/preprocessing/fargate-unzip/template.yaml

```diff
@@ -0,0 +1,311 @@
+Parameters:
+  pTaskName:
+    Type: String
+    Description: "General Task Name describing the general objective of this transformation."
+    Default: "unzipping"
+    AllowedPattern: "[a-z0-9]+?-?[a-z0-9]+?"
+  pEnv:
+    Type: String
+    Description: "Environment to deploy"
+    AllowedValues:
+      - dev
+      - test
+      - prod
+  pS3LandingBucketName:
+    Type: String
+    Description: "Landing bucket with the untreated files."
+  pS3LandingPrefix:
+    Type: String
+    Description: "Prefix of the landing bucket that will be treated"
+    Default: ""
+  pS3LandingExt:
+    Type: String
+    Description: "File extension of the files to be treated"
+    Default: ".txt.gz"
+  pS3RawBucketName:
+    Type: String
+    Description: "RAW bucket for SDLF, this bucket MUST ALREADY EXISTS"
+  pEcrUnzipDockerImage:
+    Type: String
+    Description: "ECR Image URL with the docker that will perform treating steps in the form: repository-url/image:tag"
+  pFargateTaskCPU:
+    Type: String
+    Description: "Number of cpu units used by the task"
+    AllowedValues:
+      - 256
+      - 512
+      - 1024
+      - 2048
+      - 4096
+    Default: 2048
+  pFargateTaskMemory:
+    Type: String
+    Description: "Amount of memory used by the task"
+    AllowedValues:
+      - 512
+      - 1024
+      - 2048
+      - 3072
+      - 4096
+      - 5120
+      - 6144
+      - 7168
+      - 8192
+      - 8192
+      - 30720
+    Default: 8192
+  pSubnetId:
+    Type: String
+    Description: "The name of the Subnet Id"
+  pAthenaBucketName:
+    Type: String
+    Description: "Name of the Athena Bucket"
+
+Resources:
+  ServiceLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/ecs/${AWS::StackName}-converter-task"
+    UpdateReplacePolicy: "Retain"
+    DeletionPolicy: "Retain"
+
+  #################### BUCKETS ####################
+  rS3LandingBucket:
+    Type: "AWS::S3::Bucket"
+    DeletionPolicy: "Retain"
+    UpdateReplacePolicy: "Retain"
+    Properties:
+      BucketName: !Ref pS3LandingBucketName
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: "AES256"
+      VersioningConfiguration:
+        Status: "Enabled"
+      NotificationConfiguration:
+        QueueConfigurations:
+          - Event: "s3:ObjectCreated:*"
+            Queue: !GetAtt "rLandingSQSQueue.Arn"
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: Prefix
+                    Value: !Ref pS3LandingPrefix
+                  - Name: Suffix
+                    Value: !Ref pS3LandingExt
+
+  #################### SQS ####################
+  rLandingSQSDeadLetterQueue:
+    Type: "AWS::SQS::Queue"
+
+  rLandingSQSQueue:

```

Comment:
If possible, can we also encrypt as much as possible using the existing KMS key?
https://github.com/awslabs/aws-serverless-data-lake-framework/blob/b4945543f0a8f15f3516c68323822ba4fba00a3e/sdlf-foundations/nested-stacks/template-kms.yaml#L111

Reply:
I'll use /SDLF/KMS/KeyArn key